cppcache/src/ThinClientRegion.cpp

```diff
@@ -2672,25 +2672,29 @@ GfErrType ThinClientRegion::clientNotificationHandler(TcrMessage& msg) {
       LocalRegion::tombstoneOperationNoThrow(msg.getTombstoneVersions(),
                                              msg.getTombstoneKeys());
       break;
-    default: {
-      if (TcrMessage::getAllEPDisMess() == &msg) {
+    default:
+      try {
+        auto& marker =
+            dynamic_cast<TcrMessageAllEndpointsDisconnectedMarker&>(msg);
         setProcessedMarker(false);
+        LOGDEBUG(
+            "ThinClientRegion::clientNotificationHandler: rec'd endpoints "
+            "disconnected message");
         LocalRegion::invokeAfterAllEndPointDisconnected();
-      } else {
+      } catch (std::bad_cast) {
```

Comment:
```c++
} catch (std::bad_cast&) {
```

Reply:
Yeah, LGTM griped at me about this one...